Created Trimmer Decorator & Capitalize Decorator & Base Decorator and Namable classes

OpondoG opened the pull request:
In this project, I will use the Decorator design pattern to correct the names of people.
Interface

- Create a class Nameable.

Implement a method called correct_name that will raise a NotImplementedError.
Turn your Person class to Nameable

- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
Create a class that inherits from the base Decorator class.
Implement a method correct_name that capitalizes the output of @nameable.correct_name.

- For the TrimmerDecorator:

Create a class that inherits from the base Decorator class.
Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

OpondoG committed:
Created Trimmer Capitalize Base Namable classes

Danieldotcomcoder reviewed (commented):
Hi @OpondoG , 


While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't include the new professional Readme template , Kindly refer to this [template](https://github.com/microverseinc/readme-template/blob/master/README.md) 👍 👍  to add a good Readme File 😄 😄 then submit again, Thank you 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

Kamba56 reviewed (commented):
Hi @OpondoG ,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
You have submitted a project, that doesn't include the new professional Readme template , Kindly refer to this [template](https://github.com/microverseinc/readme-template/blob/master/README.md) +1 +1 to add a good Readme File smile smile then submit again, Thank you +1

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready pray

Cheers and Happy coding!clapclapclap

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

Invalid Code Review Request does not count into the code reviews limit.

Kamba56 reviewed (commented):
Hi @OpondoG

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**Please use the new readme template provided by microverse. Check [here](https://github.com/microverseinc/readme-template/blob/master/README.md) for the template.**

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

OpondoG commented:
Hi @Kamba56 

Thank you for the code review, I would like to get a little bit of insight about the new ReadME template. Are we not supposed to modify it?

addod19 commented:
### Hi @OpondoG, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because the README file has not met the requirements as seen in the screenshot below
![Screenshot from 2023-01-06 09-55-55](https://user-images.githubusercontent.com/46388113/210976008-8f7c28a5-06ff-450f-b733-6cacd51c77da.png)
I hope this helps

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

codepantha reviewed (commented):
# STATUS: INVALID REVIEW REQUEST 🛑

Hi @OpondoG 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't use the new [README template](https://github.com/microverseinc/readme-template/blob/master/README.md). Please click [here](https://github.com/microverseinc/readme-template/blob/master/README.md) to access the new template and use it in your project 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

OpondoG commented:
> Hi @codepantha 👋,
> 
> Thank you for the code review. I just used the README that was attached I don't understand where the problem is.
>

mirouhml commented:
## Invalid Code Review Request

Hi @OpondoG,

Please know that you didn't use the right README, there are no new commits that show you did that maybe the issue is that you didn't push your code or something, kindly check that and see if you're following [this template](https://github.com/microverseinc/readme-template) with all its main headers available, also make sure to edit it and put your information there.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

OpondoG committed:
Modify README.md

OpondoG committed:
Merge branch 'dev' into decorate

OpondoG commented:
Hi @mirouhml  👋,

Thank you for the insight I just realized that the commits have not been making active changes on GitHub.

Nemwel-Boniface commented on base_decorator.rb:
```diff
@@ -0,0 +1,12 @@
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
- [x] You seem to have forgotten to have the getter and setter for nameable. Kindly include the attribute accessor for nameable. ```attr_accessor :nameable```

Nemwel-Boniface commented on captalize_decorator.rb:
```diff
@@ -0,0 +1,12 @@
+    @nameable = nameable
+  end
+
+  def correct_name
+    super.capitalize
+  end
```
- [x] For this section, you were supposed to "Implement a method correct_name that capitalizes the output of @nameable.correct_name.". This can be implemented like this: ```@nameable.correct_name.capitalize```

Nemwel-Boniface reviewed (changes requested):
## Hello @OpondoG         :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!

![](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)
## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

OpondoG committed:
Make Required changes

OpondoG committed:
Merge branch 'decorate' of https://github.com/OpondoG/OOP-School-Library into decorate

OpondoG committed:
Fix Linter Error

OpondoG commented:
Hi @Nemwel-Boniface ,🥳 

Thank you for taking the time to review my code. I really appreciate your feedback and insights.

I have addressed all of the issues you raised in my code. I hope these changes address your concerns.

If you have any further questions or concerns, please don't hesitate to let me know. I'm happy to continue this conversation and make any additional changes that may be necessary.

Thank you again for your help.

muneebulrehman reviewed (approved):
Hi @OpondoG , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
